pkg/operator/ceph/object/rgw.go

```diff
@@ -245,6 +246,11 @@ func (c *clusterConfig) deleteStore() error {
 
 		err = deleteRealmAndPools(objContext, c.store.Spec)
 		if err != nil {
+			if strings.Contains(err.Error(), "signal: interrupt") {
```

Comment:
Is `deleteRealmAndPools()` already implemented to be best effort or does it return on the first failure of trying to clean up a resource? We probably need more changes inside that method so all cleanup operations are best effort.

Reply:
`deleteRealmAndPools` has deletion methods for `single site` and `multi-site`. 
1. Single Site
 - `deleteRealm` and `deletePools` are already best effort implementation. 
 - Added a check to ignore timeout error when listing the `realm` in `getObjectStores`. 

Checking for multi site.

Reply:
I'd suggest the multi-site cleanup should also be best-effort. @alimaredia wdyt? Can you take a look and perhaps open an issue to follow up separately from this PR?

Reply:
@sp98 All the multisite teardown code is doing is just removing the object store endpoint from the zones/zonegroups and then running the same deletePools() function that single site is. If you think single site cleanup should be best effort, then multisite should be as well.